boringssl/boringssl_jni/ecrecover.c

```diff
@@ -0,0 +1,159 @@
+#include "ecrecover.h"
+
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/ecdsa.h>
+#include <openssl/evp.h>
+#include <openssl/mem.h>
+#include <string.h>
+#include <stdio.h>
+
+#define P256_KEY_LEN 65
+#define P256_COORD_LEN 32
+
+int ecrecover_r1(
+    const unsigned char message_hash[32],
+    const unsigned char signature[64],
+    int recovery_id,
+    unsigned char output_buffer[65]) {
+
+    if (message_hash == NULL || signature == NULL || output_buffer == NULL) {
+        return 1;
+    }
+
+    if (recovery_id < 0 || recovery_id > 3) {
+        return 1;
+    }
```

Comment:
this should be restricted to 0,1.  ecrecover should return uncompressed points, and (2,3) imply compressed points